### DISCUSSION TITLE
Hosting using an existing proxy manager

### DISCUSSION THREAD

**Discussion by dalekcoffee** [JeffreyCA/spleeter-web#470] [Q&A]
Hello!

Bit of a newbish question here

I have succesfully deployed the application as-is, but found that it is only accessible on port 80 on localhost, and I cannot access it on other machines on my network.

I saw another post where it was mentioned that a old release enabled the use of "DEV_WEBSERVER_PORT" but I don't quite see this in documentation or in the docker compose prebuilts for CPU

Can someone advise as to how I can go about deploying this on docker with a custom port, say, 7483 (idk random) so that I can then use my existing ngix proxy manager on a diferent host and properly map http/https traffic to this tool?

Many thanks ahead!

**Comment by JeffreyCA** [comment]
Hi, can you try setting `NGINX_PORT` to the desired port in your `.env` file?

**Comment by dalekcoffee** [comment]
I created a .env file and added NGINX_PORT=456 (just gonna go with that)


Placed it within the spleeter web folder

Do I then need to run 

docker-compose -f docker-compose.yml -f docker-compose.prod.yml -f docker-compose.prod.selfhost.yml up as-is or do I need to add it to the command?

Im not familiar with using an ENV file I apologize. Doing some research but the format looks odd

Do I run this:

docker-compose -f docker-compose.yml -f docker-compose.prod.yml -f docker-compose.prod.selfhost.yml --env-file .env up

**Reply by JeffreyCA** [reply]
Yeah just run it as is, no need to explicitly specify the `.env` file in the command

**Comment by dalekcoffee** [comment]
So this works locally

![image](https://user-images.githubusercontent.com/78516780/213942684-99d0fc67-7585-413a-a7b3-cf75d3ed48f7.png)


But when accessing from another host on LAN it seems to still fail

![image](https://user-images.githubusercontent.com/78516780/213942700-26df7cbe-1d9e-4f92-bac7-9e60f52fddc2.png)



When I setup my proxy manager to this host and port, it is accessible from WAN with domain, but once again I get an HTTP 400 response

**Reply by dalekcoffee** [reply]
![image](https://user-images.githubusercontent.com/78516780/213942857-cdd23811-fdc3-4ba6-99f2-fae750540890.png)


I also disabled firewall on the host machine out of curiosity, and here's a ping to that host with docker desktop where it shows I can see the host.

Was there some other variable needed?

And to fix, do I just need to delete all containers, volumes, images, and redeploy with appended variable?

**Reply by JeffreyCA** [reply]
Oh, so in your `.env` you'll need to also set `APP_HOST` to the host (or hosts, separated by commas) that you want the app to be accessible from, like:

`APP_HOST=192.168.1.3`

No need to include `localhost` itself. And no need to delete anything, just restart the containers afterwards

**Reply by dalekcoffee** [reply]
Ah ok, do I need to also specify my domain somewhere so it will accept the WAN connections through my ngix proxy manager or is that not necessary?

**Comment by dalekcoffee** [comment]
The above did work for accessing from another host!

That said,  I registered with ngix proxy manager using my domain and on WAN it still has an error claiming: Bad Request (400)

**Reply by dalekcoffee** [reply]
![image](https://user-images.githubusercontent.com/78516780/213951981-784186d0-5d93-4b36-8faf-8ea5de188ffb.png)

According to this it states domains or IPs, separated by commas, I did try that and re-deployed.

No luck unfortunately, still bad request when accessed by domain

My env file contains:
NGINX_PORT=456
APP_HOST=192.168.1.3, subdomain.domain.com

(Domain sanitized for privacy)

**Reply by dalekcoffee** [reply]
Self resolved! I assumed I needed a space after the comma on APP_HOST, removed same

It is now working thank you again!

**Reply by JeffreyCA** [reply]
Glad it worked!

**Comment by dalekcoffee** [comment]
Resolved!
